server.go

```diff
@@ -768,6 +783,31 @@ func (s *Server) InlineParseCustomData(msg proto.Message) error {
 	return cmsg.ParseCustomChannelData(msg)
 }
 
+// Name returns the name of this endpoint. This MUST be unique across all
+// registered endpoints.
+//
+// NOTE: This method is part of the protofsm.MsgEndpoint interface.
+func (s *Server) Name() protofsm.EndpointName {
+	return tapchannel.MsgEndpointName
+}
+
+// CanHandle returns true if the target message can be routed to this endpoint.
+//
+// NOTE: This method is part of the protofsm.MsgEndpoint interface.
+func (s *Server) CanHandle(msg protofsm.PeerMsg) bool {
+	<-s.ready
+	return s.cfg.AuxFundingController.CanHandle(msg)
```

Comment:
This checks for handling for the channel funding wire messages but not the RFQ messages. Given that this method is on the tap server I would expect it to validate all tap relevant wire messages. Should it?

What's the motivation of having these methods as part of `Server` and not as part of a sub-service?

Reply:
The main reason we don't want to handle the RFQ messages in the same way is that it would mean we couldn't run `tapd` in standalone mode anymore, only bundled through `litd` (which connects this hook).
If we use the custom message RPC of `lnd` for the RFQ messages, we are independent of the mode `tapd` is running in (standalone with remote `lnd` or bundled in `litd` in integrated mode).

So this short cut is very specific to the funding process which is only really relevant in the litd integrated mode.

To your second question: The reason this is on the main `Server` instance is the way we set up those hooks in `litd`: https://github.com/lightninglabs/lightning-terminal/pull/761/commits/03479d12759d0600d33956c5df527dae97eb29e1
The `Server` struct is what we instantiate within `tapd`, so we want it to implement all the aux interfaces directly (which it will then proxy to the correct components once the correct state is achieved).
I agree that this isn't super nice and should probably be made a bit nicer in the future. But to be honest, for now I don't really have a much better idea on how to do this, without needing to add a lot more `tapd` specific code to `litd`.